packages/cohere/src/cohere-prepare-tools.ts

```diff
@@ -11,126 +11,90 @@ export function prepareTools(
   },
 ): {
   tools:
-    | undefined
     | Array<{
-        name: string;
-        description: string | undefined;
-        parameterDefinitions: Record<
-          string,
-          {
-            required: boolean;
-            type: 'str' | 'float' | 'int' | 'bool';
-            description: string;
-          }
-        >;
-      }>;
-  force_single_step: boolean | undefined;
+        type: 'function';
+        function: {
+          name: string | undefined;
+          description: string | undefined;
+          parameters: unknown;
+        };
+      }>
+    | undefined;
+  tool_choice:
+    | { type: 'function'; function: { name: string } }
+    | 'auto'
+    | 'none'
+    | 'any'
+    | undefined;
   toolWarnings: LanguageModelV1CallWarning[];
 } {
   const tools = mode.tools?.length ? mode.tools : undefined;
   const toolWarnings: LanguageModelV1CallWarning[] = [];
 
   if (tools == null) {
-    return { tools: undefined, force_single_step: undefined, toolWarnings };
+    // TODO(shaper): Mistral references `tool_choice` below as well, look into it.
```

Comment:
Some tool choices can be simulated by e.g. removing tools. So `auto` and `none` are always possible. For `required` or `tool`, you need a way to force the LLM to make a tool call.

Reply:
I've revised to implement `auto` and `none`. I don't see a way to force a call in Cohere, so I've made `required` throw. `tool` follows what Mistral does, which appears to be:
- filter out all tools other than the one specified
- leave it to the LLM (unless Mistral actually supports `any`?) to decide whether to call it (so it's not a forced call)